### ISSUE TITLE
feat: add consistent-code-fence rule (MD048)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#198]
## Overview

Implement the `consistent-code-fence` rule (equivalent to markdownlint MD048 / remark-lint `fenced-code-marker`).

Closes part of #76.

## Problem

Markdown documents often mix backtick fences (`` ``` ``) and tilde fences (`~~~`) inconsistently. This causes visual noise and violates style guides that expect a uniform fence character across a document.

## Proposed Rule Key

`consistent-code-fence`

## Detection Logic

Scan each opening fence marker (lines starting with ` ``` ` or `~~~`). Depending on the configured `style`:

| Style | Behaviour |
|---|---|
| `consistent` (default) | The **first** fence character found in the document sets the expected style; any subsequent fence using a different character is flagged |
| `backtick` | Every fence must use backtick (`` ` ``); tilde fences are always flagged |
| `tilde` | Every fence must use tilde (`~`); backtick fences are always flagged |

**Invalid** (style: `consistent`, first fence is backtick):
```markdown
\```js
console.log("hello")
\```

~~~python
print("world")
~~~
```

**Valid** (style: `consistent`):
```markdown
\```js
console.log("hello")
\```

\```python
print("world")
\```
```

**Valid** (style: `tilde`):
```markdown
~~~js
console.log("hello")
~~~

~~~python
print("world")
~~~
```

## Fix

Change flagged fence markers to match the configured or first-seen style.

## Implementation Notes

- Rule key: `consistent-code-fence`
- Config field: `style` string (`"consistent"` | `"backtick"` | `"tilde"`), default `"consistent"`
- Default enabled with `style: "consistent"`
- Only opening fences are checked; content inside code blocks is skipped
- Reuse `openingFenceMarker` and `IsClosingFence` utilities from `internal/rule/fence.go`

### Config

**Default (`DefaultConfigJSON`):**
```json
"consistent-code-fence": { "style": "consistent" }
```

**Default (`Default()` in Go):**
```go
"consistent-code-fence": {
    Enabled:  true,
    Severity: SeverityError,
    Options:  map[string]interface{}{"style": "consistent"},
},
```

**User customisation examples:**
```json
"consistent-code-fence": { "style": "backtick" }
```
```json
"consistent-code-fence": false
```

## References

- [markdownlint MD048](https://github.com/DavidAnson/markdownlint/blob/main/doc/md048.md)
- [remark-lint fenced-code-marker](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-fenced-code-marker)

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create
- `internal/rule/consistent_code_fence.go` — `CheckConsistentCodeFence` function
- `internal/rule/consistent_code_fence_test.go` — unit tests (table-driven)
- `e2e/fixtures/consistent_code_fence_valid.md` — valid fixture
- `e2e/fixtures/consistent_code_fence_violation.md` — violation fixture
- `e2e/fixtures/config-consistent-code-fence.json` — e2e config

### Files to modify
- `internal/config/config.go` — register `consistent-code-fence` in `DefaultConfigJSON` and `Default()`
- `internal/linter/linter.go` — call `CheckConsistentCodeFence` in `collectErrors()`
- `cmd/root_bench_test.go` — exercise the rule scan path in `generateComplexMarkdown`; enable in `benchmarkConfig()` with violation-free options
- `e2e/e2e_test.go` — add e2e test cases

### Checklist
- [ ] Implement `CheckConsistentCodeFence(filename string, lines []string, offset int, style string) []LintError`
  - Parse `style` (`"consistent"`, `"backtick"`, `"tilde"`)
  - Iterate lines; use `openingFenceMarker` to detect opening fences; track current code block with `IsClosingFence`
  - In `consistent` mode, record the first fence character; flag any opener using a different character
  - In `backtick`/`tilde` mode, flag any opener using the wrong character
  - Skip lines inside fenced code blocks (content only; opener/closer are checked above)
- [ ] Write unit tests covering:
  - All-backtick document (consistent / backtick / tilde modes)
  - All-tilde document (consistent / backtick / tilde modes)
  - Mixed document — violations flagged at the correct line
  - Single fence (no violation in any mode)
  - Empty document
  - Offset handling for frontmatter
- [ ] Register rule in config (`DefaultConfigJSON` + `Default()`) with `{ "style": "consistent" }`
- [ ] Integrate in linter `collectErrors()`
- [ ] Update `generateComplexMarkdown` in `cmd/root_bench_test.go` to use only backtick fences; verify `TestBenchmarkContentIsViolationFree` passes
- [ ] Add e2e fixtures and test cases
- [ ] Run `make test-all`
